tests/test_configurations/test_manager.py

```diff
@@ -235,6 +245,48 @@ def test_agent_attribute_get_set():
     agent_config_manager.verify_private_keys(DUMMY_AEA, lambda x, y, z: None)
 
 
+def test_recursive_updates() -> None:
+    """Test recursive updates."""
+    agent_config_manager = AgentConfigManager.load(DUMMY_AEA, substitude_env_vars=True)
+    agent_config_manager.set_variable(
+        "skills.test_skill.models.scaffold.args.recursive", {"foo": "bar"}
+    )
+    value = cast(
+        Dict,
+        agent_config_manager.get_variable(
+            "skills.test_skill.models.scaffold.args.recursive"
+        ),
+    )
+
+    assert value == {"foo": "bar"}
+
+    agent_config_manager.set_variable(
+        "skills.test_skill.models.scaffold.args.recursive",
+        {"hello": "world"},
+    )
+    value = cast(
+        Dict,
+        agent_config_manager.get_variable(
+            "skills.test_skill.models.scaffold.args.recursive"
+        ),
+    )
+
+    assert value == {"hello": "world"}
+
+    agent_config_manager.set_variable(
+        "skills.test_skill.models.scaffold.args.recursive.hello",
+        "world_0",
+    )
+    value = cast(
+        Dict,
+        agent_config_manager.get_variable(
+            "skills.test_skill.models.scaffold.args.recursive"
+        ),
+    )
+
+    assert value == {"hello": "world_0"}
```

Comment:
```suggestion
@pytest.mark.parametrize("path, variable, expected", (
  ("a_path", a_value, something_expected),
  ("another_path", another_value, something_else_expected),
))
def test_recursive_updates(path, variable, expected) -> None:
    """Test recursive updates."""
    agent_config_manager = AgentConfigManager.load(DUMMY_AEA, substitude_env_vars=True)
    agent_config_manager.set_variable(
        your_path, your_value
    )
    value = cast(
        Dict,
        agent_config_manager.get_variable(
            "skills.test_skill.models.scaffold.args.recursive"
        ),
    )

    assert value == expected
```

Reply:
Here tests are not independent, they test the order too